Kubernetes Installer v2023.06.20-0 release notes (#1225)
replicatedhq/replicated-docs: docs/release-notes/rn-kubernetes-installer.md

<!--RELEASE_NOTES_PLACEHOLDER-->

## v2023.06.20-0

Released on June 20, 2023

### New Features {#new-features-v2023-06-20-0}
* Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.22.0.
* Adds [Contour add-on](https://kurl.sh/docs/add-ons/contour) version 1.25.0.
* Adds [Prometheus add-on](https://kurl.sh/docs/add-ons/prometheus) version 0.65.2-46.8.0.
* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-06-09T07-32-12Z.
* Adds [Sonobuoy add-on](https://kurl.sh/docs/add-ons/sonobuoy) version 0.56.17.
* Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.11.8.

> 

Reviewer: ```suggestion
### Improvements {#improvements-v2023-06-20-0}
* Enhance migration from weave to flannel to preserve custom IP Tables rules added
```

### Bug Fixes {#bug-fixes-v2023-06-20-0}

> * Fixes previous Flannel version by backporting the fixes introduced in the latest release v2023.06.09-0 to solve the problems faced when migrate from weave to flannel.

Reviewer: @ricardomaraschini it is missing the release notes ref to: https://app.shortcut.com/replicated/story/77628/weave-to-flannel-migration-fails-if-iptables-has-default-drop-policy-on-input-chain

Could you please add this one?

Reviewer: Also, it is missing the release notes of the fix done by @laverya : https://github.com/replicatedhq/kURL/pull/4614/files

Reviewer: ```suggestion
* Fixes all previous Flannel versions by backporting the fixes introduced in the latest release v2023.06.09-0 to solve the problems faced when migrate from weave to flannel. 
* Fixes MinIO PVC resizing race condition for versions equals and upper then `2023-06-09T07-32-12Z` .
* Fixes migration from weave to flannel to allow the installer re-try to run the migration when an error is faced.
```

> * Fixes previous Flannel version by backporting the fixes introduced in the latest release v2023.06.09-0 to solve the problems faced when migrate from weave to flannel.

Reviewer 2: ```suggestion
* Fixes issues found during Weave to Flannel migration.
```

Reviewer: @ricardomaraschini IMO we need to be a little more specific here about what was fixed so that users and us in the support knows what was fixed, backport and etc. That help us in scenarios where for example we need to suggest to users use an specific version to solve the support case.

<!--RELEASE_NOTES_PLACEHOLDER-->

## v2023.06.20-0

Released on June 20, 2023

:::important
Previously releases has a known issue that may lead the system to a critical state when you are upgrading from Weave to Flannel. This known issue was sorted out for Flannel versions equals and upper than `0.21.5` in the previous release v2023.06.09-0 and backported to all Flannel versions on this release. If you want to upgrade migrating from Weave to Flannel, please ensure that you use this or future releases. 
:::

### New Features {#new-features-v2023-06-20-0}
* Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.22.0.
* Adds [Contour add-on](https://kurl.sh/docs/add-ons/contour) version 1.25.0.
* Adds [Prometheus add-on](https://kurl.sh/docs/add-ons/prometheus) version 0.65.2-46.8.0.
* Adds [MinIO add-on](https://kurl.sh/docs/add-ons/minio) version RELEASE.2023-06-09T07-32-12Z.
* Adds [Sonobuoy add-on](https://kurl.sh/docs/add-ons/sonobuoy) version 0.56.17.
* Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) verison 1.11.8.

> ### Improvements {#improvements-v2023-06-20-0}

Reviewer: ```suggestion

### Improvements {#improvements-v2023-06-20-0}
```

Reviewer: Missing spaces

* Enhance migration from weave to flannel to preserve custom IP Tables rules added

> ### Bug Fixes {#bug-fixes-v2023-06-20-0}

Reviewer: ```suggestion

### Bug Fixes {#bug-fixes-v2023-06-20-0}
```

Missing spaces

<!--RELEASE_NOTES_PLACEHOLDER-->

## v2023.06.20-0

Released on June 20, 2023

:::important

> Previous to this release is a known issue that may put the system in a critical state when migrating from Weave to Flannel. Please ensure that you use this or future releases when migrating from Weage to Flannel. 

Reviewer: ```suggestion
Previous to this release is a known issue that may put the system in a critical state when migrating from Weave to Flannel. Please ensure that you use this or future releases when migrating from Weave to Flannel. 
```
Just a nit ^

> :::

Reviewer: @emosbaugh @kyleprice IMO we should have the above msg so that we can warn users, we can have it in the docs which help us in the support if we need to have this info and etc in order to know that is not recommended use previous versions to do this upgrade.

Reviewer: @kyleprice we backport the fix for ALL flannels versions
However, it is impossible to backport the fix for previous kURL releases

Ditto so, we need to provide the info and warn the users that they should NOT use previous kURL releases to migrate from Weave to Flannel so that they might fail in the critical scenario as we done in the past. You can check in this document all places that we have used this channel as their scenarios,
